game/configitem.cc

```diff
@@ -37,26 +37,46 @@ ConfigItem::ConfigItem(OptionList opts)
 : m_type("option_list"), m_value(opts) {
 }
 
-ConfigItem::NumericValue& ConfigItem::getMin() {
+ConfigItem::NumericValue ConfigItem::getMin() const {
 	return m_min;
 }
 
-ConfigItem::NumericValue& ConfigItem::getMax() {
```

Comment:
I don't like adding setters. I think it's unnecessary clutter for something that will be used maybe two times in the code.

Reply:
@Baklap4  @OznOg what do you think?

My arguments from discord:

I don't think so, cause the setters make the code more readable , findable and understandable. And modifying values from outside is a bad habit/style (except for pods) cause the class should be the master of its data.

Reply:
since i stepped away from java and got introduced into c# i hated getters/setters from the gecko since you're just internally setting/getting a member field.

Yes if you have to do some funky stuff like lazy loading sure that's a valid method. But for just assigning + reading i'm with @Lord-Kamina 

This is especially true with this syntax (in C#):
```csharp
public int SomeValue { get; init; } // or
public int SomeOtherVal { get; private set; }
```

The first syntax only allows the constructor to fill the `SomeValue` member whilst being publicly gettable.
The second syntax only allows for private sets whilst being publicly gettable

if something like the above is available in c++ without the boilerplate of getter/setter methods i'm more than willing to use that approach

Reply:
I don't think that can be done in C++.

Also, keep in mind that this proposed interface is due to a refactor. Originally, it was the ConfigItem itself that set these values

Reply:
That is all syntactic suggar not available in c++. There you have 
```c++
public int get_SomeValue();
public void set_SomeValue(int);
```
like in c++ COM. Found some other solutions here https://stackoverflow.com/questions/8368512/does-c11-have-c-style-properties

OK, if we do not want setters here, than I would prefer also no getters. So we come back to direct access to the members? 
@Baklap4  and @Lord-Kamina what do you think?

Reply:
Hmmm i think this gets way too complex for such a simple thing. So i think it'd be best to resort to just using member fields like we always did

I do know we have some get/set methods but those either have more stuff than just returning or setting a member field. If that's not the case though i think it'd be best to also resolve them into just member fields only